pkg/kubelet/kubeletfakes/fake_plugin_watcher.go

```diff
@@ -0,0 +1,186 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by counterfeiter. DO NOT EDIT.
+package kubeletfakes
+
+import (
+	"sync"
+
+	"k8s.io/kubernetes/pkg/kubelet/util/pluginwatcher"
+)
+
+type FakePluginWatcher struct {
+	AddHandlerStub        func(pluginType string, handler pluginwatcher.PluginHandler)
+	addHandlerMutex       sync.RWMutex
+	addHandlerArgsForCall []struct {
+		pluginType string
+		handler    pluginwatcher.PluginHandler
+	}
+	StartStub        func() error
+	startMutex       sync.RWMutex
+	startArgsForCall []struct{}
+	startReturns     struct {
+		result1 error
+	}
+	startReturnsOnCall map[int]struct {
+		result1 error
+	}
+	StopStub        func() error
+	stopMutex       sync.RWMutex
+	stopArgsForCall []struct{}
+	stopReturns     struct {
+		result1 error
+	}
+	stopReturnsOnCall map[int]struct {
+		result1 error
+	}
+	invocations      map[string][][]interface{}
+	invocationsMutex sync.RWMutex
+}
+
+func (fake *FakePluginWatcher) AddHandler(pluginType string, handler pluginwatcher.PluginHandler) {
+	fake.addHandlerMutex.Lock()
+	fake.addHandlerArgsForCall = append(fake.addHandlerArgsForCall, struct {
+		pluginType string
+		handler    pluginwatcher.PluginHandler
+	}{pluginType, handler})
+	fake.recordInvocation("AddHandler", []interface{}{pluginType, handler})
+	fake.addHandlerMutex.Unlock()
+	if fake.AddHandlerStub != nil {
+		fake.AddHandlerStub(pluginType, handler)
+	}
+}
+
+func (fake *FakePluginWatcher) AddHandlerCallCount() int {
```

Comment:
Are these `xxxCallCount` and `xxxForCall` methods used anywhere, is that for testify ?

Reply:
This is the fake, automatically generated by [counterfeiter](https://github.com/maxbrunsfeld/counterfeiter). It just generates those methods for all the methods of the interface it is meant to fake.